testsuite/mlir/mlir-parse/function.verona

```diff
@@ -1,23 +1,22 @@
 // Copyright Microsoft and Project Verona Contributors.
 // SPDX-License-Identifier: MIT
 
-// RUN: verona-mlir %s -o - | FileCheck %s
+empty_declaration();
+
+// Is this really valid?
+single_arg(x);
```

Comment:
My understanding is that this is not valid, unless x is a type name, since top level declarations should not require inference. I may be incorrect though.

Reply:
right, so I'd need to declare the type every time? I have changed the remaining tests to reflect that, but this is the `function` test, so needs to stress a bit more.

However, if I understand correctly, this is a parsing error, not a lowering error, so the AST should never have allowed this to be here in the first place.

If that's true, then I'll remove this test, but not raise an error in that case, assuming a future version of the parser will do so.

Reply:
I don't want to do interprocedural type inference.  So that is my view too.  We may relax that later for inference within a module for private members.

Reply:
Ok, I'll drop this line from the test and check all others for similar patterns.